gateway-messages/src/lib.rs

```diff
@@ -141,6 +141,66 @@ pub enum SwitchDuration {
     Forever,
 }
 
+/// Sensor readings that we could request from the target by `SensorId`
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub enum SensorRequest {
+    /// Requests the current timestamp (as a `u64`)
+    CurrentTime,
```

Comment:
This isn't specific to a sensor (or even sensors at all, except that they also return timestamps), right? Should it be a top-level `MgsRequest::CurrentTime` variant? If we did that, we could also consider splitting this enum up into a struct/enum pair to avoid repeating `id`, something like

```rust
pub enum SensorRequestKind { LastReading, LastData, LastError, ErrorCount }

pub struct SensorRequest {
    sensor_id: u32,
    kind: SensorRequestKind,
}
```

Reply:
I like this a lot, done in 6b0afefb7f232caba837f38b3e87c426e99ed561